DllImportGenerator/DllImportGenerator/MarshallingAttributeInfo.cs

```diff
@@ -48,7 +48,31 @@ CharEncoding CharEncoding
         UnmanagedType UnmanagedArraySubType,
         int ArraySizeConst,
         short ArraySizeParamIndex,
-        CharEncoding CharEncoding) : MarshallingInfoStringSupport(CharEncoding);
+        CharEncoding CharEncoding) : MarshallingInfoStringSupport(CharEncoding)
+    {
+        public MarshalAsInfo(UnmanagedType unmanagedType, CharEncoding charEncoding)
+            :this(unmanagedType, 0, 0, 0, charEncoding)
+        {
+        }
+
+        public MarshallingInfo CreateArraySubTypeMarshalAsInfo()
+        {
+            Debug.Assert(UnmanagedType is UnmanagedType.LPArray or UnmanagedType.ByValArray or UnmanagedType.SafeArray);
+            if (UnmanagedArraySubType == 0)
```

Comment:
Boo. If we are going to use the type system with nullability, we also need to use it properly and accept that `0` isn't valid here. Which means we need a locally named constant with is used to indicate this state.

Reply:
The `UnspecifiedData` below is a good example.